.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,49 @@
+# Copyright 2024 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: e2e
+
+on:
+  workflow_run:
+    workflows: [Image]
+    types:
+      - completed
+
+jobs:
+  e2e-tests:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push'
```

Comment:
We don't trigger on `push` anymore. Is this ever true?

Reply:
`push` means on merge to those branches

Reply:
Yes, I undertand that. We have now changed the `on` section to trigger on completion of the `Image` workflow. Does this still propagate the `push` event that initially triggered it?

Reply:
that's my concern and the why I am having this line, I don't know if the `workflow_run` will also trigger the `e2e` during the PR. so this is kind of a double safe

Reply:
That doesn't really answer my question. I understand the intent of setting this here. I want to know if you have seen whether this is set given the triggers that we're using? There are enough usages of `github.event == workflow_run`  (for example https://github.com/orgs/community/discussions/21090#discussioncomment-3224648) so that it seems as if the `push` comparison here will NOT do what we want.

Looking at https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_runbranchesbranches-ignore we probably want to expand the event to include a branch selector / filter.

It may also be useful to distinguish between a successful and unsuccessful completed run, but maybe that's already handled in some way.